### PR TITLE
fix: unclear message when async action is used in sync fields

### LINKED ISSUE #21028: [Bug]: Show better message with async framework action is used in a data field
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

The lint error message "storeValue is not defined" doesn't do much to inform users about the cause of the issue. This should be changed to a better clause.

### Steps To Reproduce

1. Drag a button
2. Add {{storeValue}}
3. See lint error

### Public Sample App

_No response_

### Environment

Production

### Issue video log

_No response_

### Version

Cloud

### PR DESCRIPTION
**Pull request by ohansFavour**
## Description
This PR replaces lint error message "`asyncAction` is not defined" to "Async framework action `asyncAction` cannot be executed in a function that is bound to a sync field.`"

Fixes #21028 

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?
-WIP

### Dev activity
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

### QA activity:
- [ ] Test plan has been approved by relevant developers
- [ ] Test plan has been peer reviewed by QA
- [ ] Cypress test cases have been added and approved by either SDET or manual QA
- [ ] Organized project review call with relevant stakeholders after Round 1/2 of QA
- [ ] Added Test Plan Approved label after reveiwing all Cypress test

### CODE REVIEW

File: app/client/src/globalStyles/CodemirrorHintStyles.ts [hunk 1]
```diff
@@ -270,6 +270,8 @@ export const CodemirrorHintStyles = createGlobalStyle<{
     margin-bottom: 5px;
     font-family: ${(props) => props.theme.fonts.text};
     color: #4B4848;
+    background-position: 0 2.8px;
+    padding-left: 20px;
```
[the remark this answers is not in the document]
Comment (ohansFavour) [reply app/client/src/globalStyles/CodemirrorHintStyles.ts:274]: ## DP
Screenshot 2023-03-02 at 20 40 10

## Production
Screenshot 2023-03-02 at 20 41 56

[the remark this answers is not in the document]
Comment (ohansFavour) [reply app/client/src/globalStyles/CodemirrorHintStyles.ts:274]: ## Changes

Icon close to the lint message re-aligned.

**Review by Rishabh-Rathod — APPROVED** [review]
LGTM

**Pull request merged**